src/main/scala/model/entities/Enemy.scala

```diff
@@ -2,25 +2,26 @@ package model.entities
 
 import model.common.DefaultValues
 import model.entities.TroopState.*
-import model.entities.WorldSpace.Position
+import model.entities.WorldSpace.{LanesLength, NumOfLanes, Position}
 
 import scala.concurrent.duration.FiniteDuration
+import scala.util.Random
 
 /**
- * Interface of an enemy model
+ * A Enemy is an abstract entity that models the common behaviour of different types of enemy.
+ *
+ * @param position the position in which the enemy is placed.
+ * @param life the life that the enemy currently has.
+ * @param state the state of the enemy.
  */
 abstract class Enemy(override val position: Position,
```

Comment:
Consider changing the name to Zombie and then find a new name for the basic Zombie.
The game is called Plants vs Zombies so a generic Troop should be a Plant or a Zombie, not a Plant or an Enemy.